### PR TITLE
500 errors with json are unhandled

### PR DESCRIPTION
**Pull request by hharnisc**
Related to https://github.com/bufferapp/buffer-rpc/pull/4

Marks all non-400 errors as unhandled

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,13 +23,21 @@ class RPCClient {
...
+            }))
+            .catch(error => reject(error));
+        }))
+      .then(({ response, status }) => {
         if (response.error) {
           // throw a handled exception
           const err = new Error(response.error);
           err.code = response.code;
-          err.handled = true;
+          err.handled = status === 400;
```
Comment (esclapes) [inline index.js]: Would it make sense to use `error.code` to set the `handled` property here? 

There is a [related comment about this on the `buffer-rpc`](https://github.com/bufferapp/buffer-rpc/pull/4#discussion_r223325726) pull request, if we set a specific code in our error handling middleware, we could rely on it here.

My main concern with using `status === 400` is that the [`createError` helper function](https://github.com/bufferapp/buffer-rpc/blob/master/src/createError.js#L1), allows for `statusCode` to be passed as an option but it is not clear (on call site) that using that parameter would make the error _unhandled_ on the client.

Apart from that, if we rely on `response.code` the implementation would be simpler without requiring the status to be passed along to the next `then()`. 

How does this alternative approach feel to you @hharnisc?

**Comment by hharnisc** [comment]
Hey @esclapes I've refactored the client and the backend to explicitly set the handled flag. Next step is to publish a beta version of these and get this working in `Publish`

**Comment by hharnisc** [comment]
Opened a WIP PR in publish - https://github.com/bufferapp/buffer-publish/pull/377

It builds but I'm still working out the bugs :wink:

**Comment by hharnisc** [comment]
That looks like a great conf @esclapes -- and that saying is so true :smile:

**Comment by hharnisc** [comment]
The client required a couple small tweaks to get things working -- Webpack mangled the destructured `response` argument (for json and status - https://github.com/bufferapp/micro-rpc-client/pull/3/commits/048208da53aee3aab0adaeae1f9205394f569e60#diff-168726dbe96b3ce427e7fedce31bb0bcR27).

Things are working in https://github.com/bufferapp/buffer-publish/pull/377 now :smile: 

I'm going to merge this (and server side) bump the version and publish :tada:

**Comment by philippemiguet** [comment]
That's fantastic @hharnisc 🤚 😄

**Pull request merged**